concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs

```diff
@@ -430,6 +513,28 @@ data Updates' (cpv :: ChainParametersVersion) = Updates {
         pendingUpdates :: !(PendingUpdates cpv)
     }
 
+migrateUpdates :: 
+  forall oldpv pv t m . 
+    (IsProtocolVersion pv,
+     IsProtocolVersion oldpv,
+     SupportMigration m t) =>
+    StateMigrationParameters oldpv pv -> Updates oldpv -> t m (Updates pv)
+migrateUpdates migration Updates{..} = do
+  newPendingUpdates <- migratePendingUpdates migration pendingUpdates
+  let migrateKeysCollection UpdateKeysCollection{..} =
+        UpdateKeysCollection{
+          level2Keys = migrateAuthorizations migration level2Keys,
+          ..
+          }
+  newKeyCollection <- migrateHashedBufferedRef (return . StoreSerialized . migrateKeysCollection . unStoreSerialized) currentKeyCollection
+  newParameters <- migrateHashedBufferedRef (return . StoreSerialized . migrateChainParameters migration . unStoreSerialized) currentParameters
+  return Updates {
+    currentKeyCollection = newKeyCollection,
+    currentProtocolUpdate = Null, -- TODO: Make sure this is sensible. We do always clear pending protocol updates, but maybe we don't want to do that here.
```

Comment:
This should behave the same as `Concordium.Types.UpdateQueues.getUpdates`, which is to keep the protocol update intact.

Reply:
Ok.